### PR TITLE
[Banner] Updates to Banner

### PR DESCRIPTION
**Pull request by sophschneider**
### WHY are these changes introduced?

Closes: https://github.com/Shopify/polaris-summer-editions/issues/19

### WHAT is this pull request doing?

> Note: These banner updates have been decoupled from the button updates so the buttons you see here are not reflective of the final design. The dismiss button spacing may have to be modified once the iconOnly plain buttons have been merged into `v11.x.x`
> Design review: because of above, I'm going to hold off on getting a design review for the Banner until the Buttons are merged in. issue to track this [here](https://github.com/Shopify/polaris-summer-editions/issues/81)

Adds updates to Banner, Figma [here](https://www.figma.com/file/jLLkmo9r28hiqPvf4s90E4/Polaris-Components-TESTING?type=design&node-id=20%3A21&t=0DtfsDGV6aOiQ0LC-1)

### How to 🎩
[Banners with code from this PR only](https://5d559397bae39100201eedc1-albyebzmvm.chromatic.com/?path=/story/all-components-banner--with-footer-call-to-action&globals=polarisSummerEditions2023:true)
[Storybook Banners with new Buttons](https://5d559397bae39100201eedc1-cadqnumaid.chromatic.com/?path=/story/all-components-banner--with-footer-call-to-action&globals=polarisSummerEditions2023:true)

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)

| Browser |  In page | In content |
|-|-|-|
| Pixel 7 latest Chrome | Screenshot 2023-05-29 at 3 35 59 PM | Screenshot 2023-05-29 at 3 34 28 PM |
| iphone 13 latest Safari | Screenshot 2023-05-29 at 3 40 09 PM | Screenshot 2023-05-29 at 3 40 34 PM |

- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)

| Browser |  In page | In content |
|-|-|-|
| OSX Chrome | Screenshot 2023-05-29 at 3 52 52 PM | Screenshot 2023-05-29 at 3 52 43 PM |
| OSX Safari | Screenshot 2023-05-29 at 3 51 36 PM | Screenshot 2023-05-29 at 3 51 25 PM |
| OSX Firefox | Screenshot 2023-05-29 at 3 49 20 PM | Screenshot 2023-05-29 at 3 49 05 PM |
| Windows 11 latest Edge | Screenshot 2023-05-29 at 3 42 13 PM | Screenshot 2023-05-29 at 3 41 51 PM |

- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
  - [x] Lighthouse a11y and axe (automated tests) on banner stories were 100%
  - [x] Works as expected on VoiceOver and in VoiceOver Rotor
  - [x] Works as expected with keyboard only

### CODE REVIEW

File: polaris-react/src/components/Banner/Banner.scss
Comment (kyledurand) [inline polaris-react/src/components/Banner/Banner.scss]: The `:where` is now included in the [global selector](https://github.com/Shopify/polaris/pull/9291/files) so I'm hoping we can just do:
```suggestion
  #{$se23} & {
```

File: polaris-react/src/components/Banner/Banner.scss
Comment (sophschneider) [inline polaris-react/src/components/Banner/Banner.scss:173]: Not used anywhere

File: polaris-react/src/components/Banner/components/PolarisSummerEditions2023Banner/PolarisSummerEditions2023Banner.tsx
Comment (qt314) [inline polaris-react/src/components/Banner/components/PolarisSummerEditions2023Banner/PolarisSummerEditions2023Banner.tsx:126]: note for later, we should probably accept '0' as a value for `borderRadius`. right now the only way to get no border radius is `undefined` and can't be used in the breakpoint object

File: polaris-react/src/components/Button/Button.scss
Comment (qt314) [inline polaris-react/src/components/Button/Button.scss:155]: This file's changes are temporary to show what the Banner would look like once the Button updates are merged. When merging this PR, I will revert these changes and change my base to main. This means the Banner will use pre edition buttons until Button is merged

File: polaris-tokens/src/token-groups/color.ts
Comment (sophschneider) [inline polaris-tokens/src/token-groups/color.ts:817]: The [Banner Figma](https://www.figma.com/file/jLLkmo9r28hiqPvf4s90E4/Polaris-Components-TESTING?type=design&node-id=20%3A21&t=0DtfsDGV6aOiQ0LC-1) has `color-text-info-strong` as `azure[16]`

File: polaris-react/locales/en.json
Comment (sophschneider) [inline polaris-react/locales/en.json:37]: this is hardcoded into the pre uplift banner 😬

File: polaris-react/src/components/Banner/Banner.scss
Comment (sophschneider) [inline polaris-react/src/components/Banner/Banner.scss]: I use the same `.Banner` wrapper class for both the old banner and the updated banner. That means using these focus-ring mixins. I really want to refactor these to use outlines instead but that would require some extra testing so I'm going to leave it out of scope for now

File: polaris-react/src/components/Banner/Banner.tsx
Comment (kyledurand) [inline polaris-react/src/components/Banner/Banner.tsx:82]: Do you still need this check? className is only used on the current banner which wouldn't be rendered if this was false

File: polaris-react/src/components/Banner/Banner.tsx
Comment (kyledurand) [inline polaris-react/src/components/Banner/Banner.tsx]: Can we name this BannerExperimental or something?

File: polaris-react/locales/en.json
Comment (kyledurand) [reply polaris-react/locales/en.json:37]: 🤯 Can you also update the dismissButton in the current banner so we start getting the translations eventually? Maybe just a quick PR against main 🙏

File: polaris-tokens/src/token-groups/shadow.ts
Comment (kyledurand) [inline polaris-tokens/src/token-groups/shadow.ts:63]: Do we need a new token for this or can it be mapped to `shadow-md` as  `valueExperimental`?

File: polaris-react/src/components/Banner/components/PolarisSummerEditions2023Banner/utilities.ts
Comment (kyledurand) [inline polaris-react/src/components/Banner/components/PolarisSummerEditions2023Banner/utilities.ts]: Nit, this fill looks like it can be hard coded and this svg could be moved to a const

**Review by kyledurand — APPROVED** [review]
This looks great 👏

File: polaris-tokens/src/token-groups/shadow.ts
Comment (sophschneider) [reply polaris-tokens/src/token-groups/shadow.ts:63]: I saw Sara and Bernardo cleaned up the shadow tokens so I'll take a look/ask them!

File: polaris-react/src/components/Banner/Banner.tsx
Comment (sophschneider) [reply polaris-react/src/components/Banner/Banner.tsx:82]: yeah I still use the outer `.Banner` class and div so it recolors all the backgrounds and borders depending on the status. I could go into each of the status classes and override them but this seemed easier!

File: polaris-tokens/src/token-groups/color.ts
Comment (sophschneider) [reply polaris-tokens/src/token-groups/color.ts:817]: @sarahill just want to double check if this is correct

azure[14] = rgba(0, 82, 124, 1)
azure[16] = rgba(0, 33, 51, 1)

I'm pretty sure `color-text-info-strong` should be 16 instead of 14 but just double checking

File: polaris-react/src/components/Banner/Banner.scss
Comment (sophschneider) [inline polaris-react/src/components/Banner/Banner.scss:9]: I already tested these changes in https://github.com/Shopify/polaris/pull/8780/files. They had to be reverted due to a stale branch but the code itself was good. I double checked in all browsers with this PR to make sure

File: polaris-tokens/src/token-groups/color.ts
Comment (sarahill) [reply polaris-tokens/src/token-groups/color.ts:817]: Yea that's correct. Should be azure[16] instead of [14]

**Comment by sarahill** [comment]
@sophschneider the screen shots look great but noticed some missing colors for warning when looking in storybook. I know you mentioned holding off on a design review so maybe these are not ready yet?

Also looked like shadows were missing but I know Bernardo is actively working on those. 

image

**Comment by sophschneider** [comment]
@sarahill thats weird it shouldn't look like that! what storybook link are you using? the latest for this pr should be [this one](https://5d559397bae39100201eedc1-llxmmgvvtg.chromatic.com/?path=/story/all-components-banner--warning&globals=polarisSummerEditions2023:true) which looks like this. Maybe try opening in an incognito tab?

Screenshot 2023-05-30 at 6 14 37 PM

there's also [this storybook](https://5d559397bae39100201eedc1-llxmmgvvtg.chromatic.com/?path=/story/all-components-banner--warning&globals=polarisSummerEditions2023:true) if you want to see what it looks like with updated buttons

This PR should have all correct colors and shadows, only missing button updates

**Comment by sarahill** [comment]
Thanks that link worked! I was running it locally so it maybe it was something on my end

**Review by sarahill — APPROVED** [review]
This looks great 👏  I left a couple comments in the github issue based on questions you had but what you have here looks great. I was looking a little bit into the banner with no title variant but we can just follow up with that if needed in a separate PR.

**Pull request merged**
